howso/hierarchy.amlg

```diff
@@ -5,14 +5,26 @@
 	;the currently contained hierarchy as a nested assoc with (false) for trainees that are stored independently.
 	;{read_only (true) idempotent (true)}
 	#get_hierarchy
-	(call !Return (assoc payload (call !GetHierarchy) ))
+	(declare
+		;returns {
+		; 	type "assoc"
+		; 	description "The map showing the full hierarchy of the Trainee and all of its subtrainees."
+		; }
+		(assoc)
+		(call !Return (assoc payload (call !GetHierarchy) ))
+	)
 
 	;Returns the full entity path to a child trainee provided its unique trainee id if it is contained in the hierarchy.
 	;Iterates down the hierarchy searching for a trainee that matches the specified id, returns null if not found or
 	;a string error if found but trainee is stored externally as an independent trainee.
 	;{read_only (true) idempotent (true)}
 	#get_entity_path_by_id
 	(declare
+		;returns {
+		; 	type ["list" "string" "null"]
+		; 	values "string"
+		; 	description "The full entity path to the child trainee, a string describing an error, or null if the child trainee was not found"
```

Comment:
The return string could be an error message? Instead of using the standard errors key outside of payload?

Reply:
unfortunately accurate